Add Solr operator site

HoustonPutman opened the pull request:
The site really needs a first apache release (v0.3.0), so it is designed around that being available. Some of the links won't work until that is available.

It currently looks A LOT like the Solr website, I would like to change the colors eventually and possibly have a separate (but related) logo.

janhoy commented:
Also, I am unsure whether the project needs two SECURITY pages - perhaps the main one can also cover CVEs for the operator, the Dockerfile etc?

A separate NEWS section is perhaps justified for operator, but i'm not 100% convinced? Like the news about operator donated to Solr project seems to me a better fit on the main Solr News page.

HoustonPutman commented:
> I think this "OPERATOR" menu option should come before the download button?

Yeah, I had a hard time with this. I can put it before the Download button.

> Once on the sub site....

Completely agree here. Definitely think we should use a different color. Not sure about stacking the menu's though. If you had an idea for that, I'd like to see it for sure!

Will play around with the colors today though and see if something sticks. 

> Also, I am unsure whether the project needs two SECURITY pages

Fair enough, easily removed.

> A separate NEWS section is perhaps justified for operator, but i'm not 100% convinced?

I think it warrants its own news section, but I could be convinced otherwise. I foresee entries for compatibility warnings with new Kubernetes versions, and things like that.

janhoy commented:
This is one way - different color and a button to the right to get you back. In this screenshot I also removed the SECURITY, PROJECT and DOWNLOAD menus, then it starts to be more distinct already. I'm not suggesting that green color, but just an example for my thinking... And I'm not a designer :)
<img width="1193" alt="Skjermbilde 2021-03-19 kl  21 47 57" src="https://user-images.githubusercontent.com/409128/111840506-c8fb9880-88fc-11eb-9eb5-05edc4db9d6a.png">

HoustonPutman commented:
Ok, I've changed the color to an exact mix of the Solr Color and the Kubernetes Color, and I think it works quite well.

I also added the SOLR TLP button, and removed some things from the menu. For the security, so far I just removed the button. Need some more effort to remove all the places that had the new security feed.

Still think we should have a related but separate Logo for the project, will think on that a bit.

<img width="1095" alt="Screen Shot 2021-03-23 at 1 41 07 PM" src="https://user-images.githubusercontent.com/3376422/112192995-d60ad700-8bdd-11eb-9d50-cf9d42f1eea4.png">

janhoy commented:
Here you go :) I'm not sure it is advisable to rip it off even if the k8s logo is CCBY licensed :)
![SolrOperator-logo](https://user-images.githubusercontent.com/409128/112282008-f8a4fa80-8c86-11eb-8dd9-f1d367037c74.png)
Do you have other ideas?

janhoy commented on content/pages/operator/community.md:
```diff
@@ -0,0 +1,81 @@
+The Solr Operator shares mailing lists with its parent project, Apache Solr.
+Available lists can be [found here]({filename}/pages/community.md#mailing-lists-chat).
+
+#### Mailing list etiquette
```
Same here - no need to duplicate here?

janhoy commented on content/pages/operator/community.md:
```diff
@@ -0,0 +1,81 @@
+
+## Mailing Lists & Chat ##
+
+We use [mailing lists](https://apache.org/foundation/mailinglists.html) for all official project communications.
```
No need to duplicate this paragraph, as long as you link to the main mailing list page for Solr?

janhoy commented on themes/solr/templates/operator/downloads.html:
```diff
@@ -0,0 +1,160 @@
+
+    <li>CRDs:
+      <ul>
+        <li><a href="{{ SITEURL }}/operator/downloads/crds/{{ SOLR_OPERATOR_LATEST_RELEASE }}/all.yaml">All</a></li>
```
Should the CRDs live on the website, or will it be in a location "softlinked" to somewhere else?

HoustonPutman replied:
It's redirected to the same place as the helm charts, nightlies.apache.org (hopefully they will have a better location in the future.) The files do not live in the site.

HoustonPutman replied:
https://github.com/apache/solr-site/blob/main/themes/solr/templates/htaccess.template#L85

janhoy commented on themes/solr/templates/operator/downloads.html:
```diff
@@ -0,0 +1,160 @@
+    </li>
+
+    <!-- TODO: Change log -->
+    <li><a href="{{ SITEURL }}/{{ SOLR_LATEST_RELEASE | replace(".", "_") }}/changes/Changes.html">Change log</a></li>
```
Simly link to https://github.com/apache/solr-operator/releases ?

HoustonPutman replied:
Maybe so. I'm unsure of how to deal with the change list yet. We also need to generate a single bulleted changeList for the artifactHub page.

janhoy replied:
I see. So as a start link to that releases page, and then change it once we have a better location? Or could artifactHub serve as the main changelog? Guess we need changes info in the git repo or other ASF controlled location as well.

HoustonPutman replied:
Yeah, I'll go with the releases page until we have another solution.

janhoy commented on themes/solr/templates/operator/downloads.html:
```diff
@@ -0,0 +1,160 @@
+    <a class="headerlink" href="#past-versions" title="Permanent link">&para;</a>
+  </h2>
+
+  <p>Archives for all past versions of Solr are available at <a href="https://archive.apache.org/dist/solr/operator">the Apache archives</a>. See the <a href="https://github.com/apache/solr-operator/releases">Github releases for pre-v0.3.0 releases</a></p>
```
"past version of Solr-Operator"...
Is it even relevant to mention releases from pre-0.3 here?

HoustonPutman replied:
Not sure, just thought I'd include it in case people get confused.

janhoy commented on content/pages/operator/community.md:
```diff
@@ -0,0 +1,81 @@
+You may alternatively choose to clone apache's git mirror at `https://gitbox.apache.org/repos/asf?p=solr-operator.git`.
+
+## Powered By ##
+TODO: Either remove or find a list of Powered By
```
Remove the entire section for now - add later if we add one

janhoy commented on themes/solr/templates/operator/features.html:
```diff
@@ -0,0 +1,201 @@
+        <div class="box">
+          <div class="img"><img src="{{ SITEURL }}/theme/images/Solr_Icons_an_ajax_based.svg"/></div>
+          <div class="content">
+            <h3>Multiple Storage Options</h3>
```
There is a mouse-over effect for each of the feature blocks. But no link.

Could it be an idea to link to the part of the github Wiki documentation for each of the features?

HoustonPutman replied:
Yes, I think that would be a great idea.

janhoy reviewed (commented):
Some more comments.

Perhaps this PR should also add a new section to Solr's frontpage, telling the world about the Solr operator. It could be a "marketing" kind of text, designed to let people know that it is there. And it would link to the sub-site.

Also, still needs to move the "SOLR OPERATOR" menu to the left of the Download button on the TLP page?

janhoy commented:
Some other operator logos
<img height=100 src="https://pbs.twimg.com/media/D8OJh8EXkAAaMjI.jpg"/><img height=100 src="https://amadeusitgroup.github.io/Redis-Operator/imgs/logo.png"/><img height=100 src="https://www.scylladb.com/wp-content/uploads/scylla-operator@3x.png"/><img height=100 src="https://prometheus-operator.dev/logo.svg"/><img height=100 src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQyLH7erGGb0JHo3MD0ITlim_vFmb2cbwkucw&usqp=CAU"/><img height=100 src="https://sweetops.com/wp-content/uploads/2019/01/istio_operator_logo.png"/>

HoustonPutman commented:
Wow nice research. I had an idea of using the kubernetes logo and turning it into sun beams, to mimmick the solr logo. My girlfriend offered to help me mock something up. So we'll see where that goes 😃

janhoy commented:
> Wow nice research. I had an idea of using the kubernetes logo and turning it into sun beams, to mimmick the solr logo. My girlfriend offered to help me mock something up. So we'll see where that goes 😃

A rudder is not that different from sun-beams so should be room for some creativity here. Guess the hexagon shape and the rudder are elements that would hint to k8s. Then you also have the spark from the operator framework logo... A spark from the sun???
<img height=100 src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT-r2aDPByMjFSEh-CNKGTbwmRKNtzyMlkUiIC4my9uQHqq2sxInzCtIAm0x35PxXTs4A&usqp=CAU"/>

janhoy commented:
![Skjermbilde 2021-03-24 kl  17 00 18](https://user-images.githubusercontent.com/409128/112342357-72a6a500-8cc2-11eb-95d0-49b2c43e2d9e.png)

janhoy commented:
> That's actually really nice, though it might be weird since we use kubebuilder, not the operator framework.

Haha - true. The two are slowly merging to become one, aren't they? But better stick with k8s elements I guess :) We don't need to have a separate logo for the first versin of the sub page though..

HoustonPutman commented:
@janhoy, what is the typeface that Solr uses? It looks like you've used it since you added "Operator" in the same lettering.

janhoy commented:
Check out the style guide at https://solr.apache.org/theme/images/identity/Solr_Styleguide.pdf

janhoy commented:
I merged main branch into your feature branch and tested the site. The security banner is not triggered on the Operator site, probably because it is looking for solr/security news in the 'articles' variable, which holds only operator news items :) 
![Skjermbilde 2021-04-29 kl  12 00 25](https://user-images.githubusercontent.com/409128/116534491-e2700700-a8e2-11eb-92b4-e75dc61046e9.png)

HoustonPutman commented:
Weirdly I tried it locally and it looked good to go. I notice that the "Operator" menu item is still there for you, so I'm not sure you are using the latest from my branch.

HoustonPutman commented:
Do you think this is good to go? (Seeing as the release will be announced tomorrow hopefully)

janhoy reviewed (approved):
Tested again, and I must have had a stale branch, cause now it all lookg good, including the move of the "solr operator" menu item. And I see the security banner.
+1 to release

janhoy commented:
> Perhaps this PR should also add a new section to Solr's frontpage, telling the world about the Solr operator. It could be a "marketing" kind of text, designed to let people know that it is there. And it would link to the sub-site.

This was an earlier comment I had. Any thoughts? Would be nice with some k8s graphics to draw attention to the sub-project.

HoustonPutman commented:
Sorry for not responding to that, I think it would be a very good thing to add! I'm fine waiting to merge if you want to include that first, but I also think it can be done separately.

janhoy commented:
It can definitely be done separately - not indending to hold up the release, nor to sit up all night with photoshop :) Also makes sense to wait for the logo vote, and then promote it some more. Also perhaps collaborate with ASF on an official blog post about the new project.

HoustonPutman commented:
> Also makes sense to wait for the logo vote, and then promote it some more. Also perhaps collaborate with ASF on an official blog post about the new project.

Very much agree with both.
